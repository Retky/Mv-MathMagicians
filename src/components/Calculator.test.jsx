import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Calc from './Calculator';

describe('Calculator component', () => {
  it('renders "Let\'s do some math"', () => {
    // Arrange
    render(<Calc />);

    // Assert
    const headingElement = screen.getByText("Let's do some math", {
      exact: false,
    });
    expect(headingElement).toBeInTheDocument();
  });

  it('renders component to match the previous snapshot', () => {
    // Assert
    const tree = renderer.create(<Calc />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render component when button click', () => {
    // Arrange
    render(<Calc />);

    // Act
    const buttonElement = screen.getAllByRole('button');
    buttonElement.forEach((e) => userEvent.click(e));
  });
});
