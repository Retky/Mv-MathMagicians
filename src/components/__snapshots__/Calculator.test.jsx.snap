// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders component to match the previous snapshot 1`] = `
<div
  className="mainCont"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calc"
  >
    <div
      className="calcScreen"
    >
      0
    </div>
    <div
      className="calcKeys"
    >
      <button
        className="calcBtn"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calcBtn"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calcBtn"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calcBtn orangeBtn"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calcBtn"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calcBtn"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calcBtn"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calcBtn orangeBtn"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calcBtn"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calcBtn"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calcBtn"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calcBtn orangeBtn"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calcBtn"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calcBtn"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calcBtn"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calcBtn orangeBtn"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calcBtn zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calcBtn"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calcBtn orangeBtn"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
